opt_adapt/opt.py

```diff
@@ -404,50 +549,50 @@ def minimise(
             if params.disp > 0:
                 pprint(term_msg + "gtol convergence")
             break
+        # For some situation, convergence should be true, but don't satisfy the above condition
+        elif np.abs(J - op.J_progress[-2]) < params.qoi_rtol * op.J_progress[-2]:
+            if fd.norm(dJ) / dJ_init < 1e-3:
```

Comment:
Could you add a parameter to the parameter class that controls the tolerance here, rather than hard-coding 1e-3?